app/src/main/java/org/wikipedia/page/Namespace.kt

```diff
@@ -0,0 +1,223 @@
+    FORM(106),
+    FORM_TALK(107),
+    BOOK(108),
+    BOOK_TALK(109),
+    FORUM(110),
+    FORUM_TALK(111),
+    DRAFT(118),
+    DRAFT_TALK(119),
+    USER_GROUP(160),
+    ACL(162),
+    FILTER(170),
+    FILTER_TALK(171),
+    USER_WIKI(200),
+    USER_WIKI_TALK(201),
+    USER_PROFILE(202),
+    USER_PROFILE_TALK(203),
+    ANNOTATION(248),
+    ANNOTATION_TALK(249),
+    PAGE(250),
+    PAGE_TALK(251),
+    INDEX(252),
+    INDEX_TALK(253),
+    MATH(262),
+    MATH_TALK(263),
+    WIDGET(274),
+    WIDGET_TALK(275),
+    JS_APPLET(280),
+    JS_APPLET_TALK(281),
+    POLL(300),
+    POLL_TALK(301),
+    COURSE(350),
+    COURSE_TALK(351),
+    MAPS_LAYER(420),
+    MAPS_LAYER_TALK(421),
+    QUIZ(430),
+    QUIZ_TALK(431),
+    EDUCATION_PROGRAM(446),
+    EDUCATION_PROGRAM_TALK(447),
+    BOILERPLATE(450),
+    BOILERPLATE_TALK(451),
+    CAMPAIGN(460),
+    CAMPAIGN_TALK(461),
+    SCHEMA(470),
+    SCHEMA_TALK(471),
+    JSON_CONFIG(482),
+    JSON_CONFIG_TALK(483),
+    GRAPH(484),
+    GRAPH_TALK(485),
+    JSON_DATA(486),
+    JSON_DATA_TALK(487),
+    NOVA_RESOURCE(488),
+    NOVA_RESOURCE_TALK(489),
+    GW_TOOLSET(490),
+    GW_TOOLSET_TALK(491),
+    BLOG(500),
+    BLOG_TALK(501),
+    USER_BOX(600),
+    USER_BOX_TALK(601),
+    LINK(700),
+    LINK_TALK(701),
+    TIMED_TEXT(710),
+    TIMED_TEXT_TALK(711),
+    GIT_ACCESS_ROOT(730),
+    GIT_ACCESS_ROOT_TALK(731),
+    INTERPRETATION(800),
+    INTERPRETATION_TALK(801),
+    MUSTACHE(806),
+    MUSTACHE_TALK(807),
+    JADE(810),
+    JADE_TALK(811),
+    R(814),
+    R_TALK(815),
+    MODULE(828),
+    MODULE_TALK(829),
+    SECURE_POLL(830),
+    SECURE_POLL_TALK(831),
+    COMMENT_STREAM(844),
+    COMMENT_STREAM_TALK(845),
+    CN_BANNER(866),
+    CN_BANNER_TALK(867),
+    GRAM(1024),
+    GRAM_TALK(1025),
+    TRANSLATIONS(1198),
+    TRANSLATIONS_TALK(1199),
+    GADGET(2300),
+    GADGET_TALK(2301),
+    GADGET_DEFINITION(2302),
+    GADGET_DEFINITION_TALK(2303),
+    TOPIC(2600);
+
+    /** Warning: this method returns an English translation for the current namespace.  */
+    @Deprecated("")
+    fun toLegacyString(): String? {
+        var string = if (this === MAIN) null else name
+        if (string != null) {
+            string = string.toLowerCase(Locale.ENGLISH).capitalize(Locale.getDefault())
+        }
+        return string
+    }
+
+    override fun code(): Int {
+        return code
+    }
+
+    fun special(): Boolean {
+        return this === SPECIAL
+    }
+
+    fun user(): Boolean {
+        return this === USER
+    }
+
+    fun userTalk(): Boolean {
+        return this === USER_TALK
+    }
+
+    fun main(): Boolean {
+        return this === MAIN
+    }
+
+    fun file(): Boolean {
+        return this === FILE
+    }
+
+    open fun talk(): Boolean {
+        return code and TALK_MASK == TALK_MASK
+    }
+
+    companion object {
+        @JvmStatic
+        val CODE_ENUM: CodeEnum<Namespace> = object : CodeEnum<Namespace> {
```

Comment:
Very minor but we can add `fun` to the start of the `CodeEnum` interface to enable the `lambda`.

```
fun interface CodeEnum<T> {
    fun enumeration(code: Int): T
}
```

Reply:
nice!